pkg/autodiscovery/providers/prometheus_pods_test.go

```diff
@@ -0,0 +1,413 @@
+					InitConfig:    integration.Data("{}"),
+					Instances:     []integration.Data{integration.Data("{\"prometheus_url\":\"http://%%host%%:%%port%%/metrics\",\"namespace\":\"\",\"metrics\":[\"*\"]}")},
+					Provider:      names.Prometheus,
+					Source:        "kubelet:foo-ctr2-id",
+					ADIdentifiers: []string{"foo-ctr2-id"},
+				},
+			},
+		},
+		{
+			name: "multi containers, match one container",
+			check: &PrometheusCheck{
+				AD: &ADConfig{
+					KubeContainerNames: []string{"foo-ctr1"},
+				},
+			},
+			pod: &kubelet.Pod{
+				Metadata: kubelet.PodMetadata{
+					Name:        "foo-pod",
+					Annotations: map[string]string{"prometheus.io/scrape": "true"},
+				},
+				Status: kubelet.Status{
+					Containers: []kubelet.ContainerStatus{
+						{
+							Name: "foo-ctr1",
+							ID:   "foo-ctr1-id",
+						},
+						{
+							Name: "foo-ctr2",
+							ID:   "foo-ctr2-id",
+						},
+					},
+				},
+			},
+			want: []integration.Config{
+				{
+					Name:          "openmetrics",
+					InitConfig:    integration.Data("{}"),
+					Instances:     []integration.Data{integration.Data("{\"prometheus_url\":\"http://%%host%%:%%port%%/metrics\",\"namespace\":\"\",\"metrics\":[\"*\"]}")},
+					Provider:      names.Prometheus,
+					Source:        "kubelet:foo-ctr1-id",
+					ADIdentifiers: []string{"foo-ctr1-id"},
+				},
+			},
+		},
+		{
+			name: "container name mismatch",
+			check: &PrometheusCheck{
+				AD: &ADConfig{
+					KubeContainerNames: []string{"bar"},
+				},
+			},
+			pod: &kubelet.Pod{
+				Metadata: kubelet.PodMetadata{
+					Name:        "foo-pod",
+					Annotations: map[string]string{"prometheus.io/scrape": "true"},
+				},
+				Status: kubelet.Status{
+					Containers: []kubelet.ContainerStatus{
+						{
+							Name: "foo-ctr",
+							ID:   "foo-ctr-id",
+						},
+					},
+				},
+			},
+			want: []integration.Config{},
+		},
+		{
+			name: "container name regex",
+			check: &PrometheusCheck{
+				AD: &ADConfig{
+					KubeContainerNames: []string{"bar", "*o-c*"},
+				},
+			},
+			pod: &kubelet.Pod{
+				Metadata: kubelet.PodMetadata{
+					Name:        "foo-pod",
+					Annotations: map[string]string{"prometheus.io/scrape": "true"},
+				},
+				Status: kubelet.Status{
+					Containers: []kubelet.ContainerStatus{
+						{
+							Name: "foo-ctr",
+							ID:   "foo-ctr-id",
+						},
+					},
+				},
+			},
+			want: []integration.Config{
+				{
+					Name:          "openmetrics",
+					InitConfig:    integration.Data("{}"),
+					Instances:     []integration.Data{integration.Data("{\"prometheus_url\":\"http://%%host%%:%%port%%/metrics\",\"namespace\":\"\",\"metrics\":[\"*\"]}")},
+					Provider:      names.Prometheus,
+					Source:        "kubelet:foo-ctr-id",
+					ADIdentifiers: []string{"foo-ctr-id"},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.check.init()
+			configs := tt.check.configsForPod(tt.pod)
+			if !reflect.DeepEqual(configs, tt.want) {
```

Comment:
Any reason not to use `assert.Equal` here (it usually produces a very helpful diff).

Reply:
or with github.com/google/go-cmp